### PR TITLE
Properly clear JIT cache on save states

### PR DESCRIPTION
**Pull request by RisingFog**
Fixes issue 8738: https://bugs.dolphin-emu.org/issues/8738

Old behaviour only cleared the block cache on save states, but not unordered maps.

Thanks to hk.konpie for his patch which I used in part to resolve the issue and skid_au for pointing me in the right direction where the JIT was clearing the cache on DoState().

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3535)

### CODE REVIEW

File: Source/Core/DolphinWX/FrameTools.cpp [hunk 1]
```diff
@@ -541,7 +541,6 @@ void CFrame::PopulateToolbar(wxToolBar* ToolBar)
 		h = m_Bitmaps[Toolbar_FileOpen].GetHeight();
 	ToolBar->SetToolBitmapSize(wxSize(w, h));
 
-
```
Comment (lioncash) [inline Source/Core/DolphinWX/FrameTools.cpp]: This FrameTools change is unrelated.

File: Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp [hunk 2]
```diff
@@ -603,8 +603,12 @@ void Wiimote::DoState(PointerWrap& p)
 		if (p.mode == PointerWrap::MODE_READ)
 		{
 			//clear
-			while (!m_read_requests.empty())
+			while (!m_read_requests.empty())
```
Comment (lioncash) [inline Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp]: Remove the space after the ending )

**Comment by delroth** [comment]
Reviewed 5 of 7 files at r1, 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp, line 608 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535#-K8RV7gdqAfU_DFiRgZe:-K8RV7geaOI9d81MkSes:1011938367) ([raw file](https://github.com/dolphin-emu/dolphin/blob/346c14b7c8cf5d0bdf8db1811e13fbc51d4e4412/Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp#L608)):*
Can we make that a vector instead? We're keeping size+data anyway.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by RisingFog** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp, line 608 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535#-K8RV7gdqAfU_DFiRgZe:-K8RXV_-1M2e2-1GF31c:1584746450) ([raw file](https://github.com/dolphin-emu/dolphin/blob/346c14b7c8cf5d0bdf8db1811e13fbc51d4e4412/Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp#L608)):*
Not sure the proper way to use a vector in this situation, but I'll try to see what I can do

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by Tilka** [comment]
Reviewed 5 of 7 files at r1, 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by delroth** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp, line 608 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535#-K8RV7gdqAfU_DFiRgZe:-K8RYDiZwDG2_Y1iPXPo:-1317930439) ([raw file](https://github.com/dolphin-emu/dolphin/blob/346c14b7c8cf5d0bdf8db1811e13fbc51d4e4412/Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp#L608)):*
I would basically replace both the size and data attributes with std::vector<u8>. There needs to be a small amount of changes for the state loading (load size to a temporary variable, use it to resize() the vector, then load into the vector .data()) but I don't see major blocking issues.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by RisingFog** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp, line 608 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535#-K8RV7gdqAfU_DFiRgZe:-K8Rtj6eEJbzywy9gT7F:-1714649596) ([raw file](https://github.com/dolphin-emu/dolphin/blob/346c14b7c8cf5d0bdf8db1811e13fbc51d4e4412/Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp#L608)):*
After looking into it more, it gets pretty deep within the Emulated Wiimote code. Lioncash has said in IRC he'll follow up with cleanup/conversion.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by RisingFog** [comment]
Reviewed 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by Sonicadvance1** [comment]
Reviewed 1 of 7 files at r1, 1 of 2 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp, line 608 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535#-K8RV7gdqAfU_DFiRgZe:-K8Ur2E4GXze-1d8Boby:-2117724249) ([raw file](https://github.com/dolphin-emu/dolphin/blob/346c14b7c8cf5d0bdf8db1811e13fbc51d4e4412/Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp#L608)):*
If @lioncash is going to come along and clean this up then I don't see a reason not to merge it as is atm.h

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*

**Comment by Tilka** [comment]
Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3535)*
